src/framework/xr/xr-views.js

```diff
@@ -0,0 +1,196 @@
+     * and are added mid-session. They can be added/removed mid session by underlyng system.
+     *
+     * @event XrViews#add
+     * @param {XrView} view - XrView that has been added.
+     * @example
+     * xr.views.on('add', function (view) {
+     *     // view that has been added
+     * });
+     */
+
+    /**
+     * Fired when view has been removed. They can be added/removed mid session by underlyng system.
+     *
+     * @event XrViews#remove
+     * @param {XrView} view - XrView that has been removed.
+     * @example
+     * xr.views.on('remove', function (view) {
+     *     // view that has been added
+     * });
+     */
+
+    /**
+     * An array of {@link XrView}'s of this session. Views are not available straight
+     * away on session start, and can be added/removed mid-session. So use of add/remove
+     * events is required for accessing views.
+     *
+     * @type {XrView[]}
+     */
+    get list() {
+        return this._list;
+    }
+
+    /**
+     * How many views are available. Views can be added/removed mid-session by underlying
+     * WebXR system.
+     *
+     * @type {number}
+     */
+    get size() {
+        return this._list.length;
+    }
+
+    /**
+     * Check if Camera Color is supported. It might be still unavailable even if requested,
+     * based on hardware capabilities and granted permissions.
+     *
+     * @type {boolean}
+     */
+    get supportedColor() {
+        return this._supportedColor;
+    }
+
+    /**
+     * Check if Camera Color is available. This information becomes available only after
+     * session has started.
+     *
+     * @type {boolean}
+     */
+    get availableColor() {
+        return this._availableColor;
+    }
+
+    /**
+     * @param {*} frame - XRFrame from requestAnimationFrame callback.
+     * @param {XRView} xrView - XRView from WebXR API.
+     * @ignore
+     */
+    update(frame, xrViews) {
+        for (let i = 0; i < xrViews.length; i++) {
+            this._indexTmp.set(xrViews[i].eye, xrViews[i]);
+        }
+
+        for (const [eye, xrView] of this._indexTmp) {
+            let view = this._index.get(eye);
+
+            if (!view) {
+                // add new view
+                view = new XrView(this._manager, xrView);
+                this._index.set(eye, view);
+                this._list.push(view);
+                view.update(frame, xrView);
+                this.fire('add', view);
+            } else {
+                // update existing view0
+                view.update(frame, xrView);
+            }
+        }
+
+        // remove views
+        for (const [eye, view] of this._index) {
+            if (this._indexTmp.has(eye))
+                continue;
+
+            view.destroy();
+            this._index.delete(eye);
+            const ind = this._list.indexOf(view);
+            if (ind !== -1) this._list.splice(ind, 1);
+            this.fire('remove', view);
```

Comment:
should you fire the event before destroying the view? Otherwise the user gets destroyed view.

Reply:
ok, keep consistent then.